broadlink/__init__.py

```diff
@@ -166,28 +166,31 @@ def __init__(self, host, mac, devtype, timeout=10):
             self.update_aes = self.update_aes_crypto
 
         self.aes = None
-        key = bytearray(
+        self.key = bytearray(
             [0x09, 0x76, 0x28, 0x34, 0x3f, 0xe9, 0x9e, 0x23, 0x76, 0x5c, 0x15, 0x13, 0xac, 0xcf, 0x8b, 0x02])
-        self.update_aes(key)
 
     def update_aes_pyaes(self, key):
         self.aes = pyaes.AESModeOfOperationCBC(key, iv=bytes(self.iv))
 
     def encrypt_pyaes(self, payload):
+        self.update_aes(self.key)
         return b"".join([self.aes.encrypt(bytes(payload[i:i + 16])) for i in range(0, len(payload), 16)])
 
     def decrypt_pyaes(self, payload):
+        self.update_aes(self.key)
         return b"".join([self.aes.decrypt(bytes(payload[i:i + 16])) for i in range(0, len(payload), 16)])
 
     def update_aes_crypto(self, key):
         self.aes = Cipher(algorithms.AES(key), modes.CBC(self.iv),
                           backend=default_backend())
 
     def encrypt_crypto(self, payload):
+        self.update_aes(self.key)
```

Comment:
We should not reinitialize aes for crypto each time.

Reply:
Hi Daniel,

Thanks for taking a look at this.

That is the exact fix we are making here, since the device seems not to be accepting later packets encrypted in CBC mode. It works fine if you reinitialise the object to go back to the beginning of the chain.

I'm happy to do this differently, if you can suggest an alternative. This change is basically just reverting to the original approach from 0.10 and before where the object was reinitialised on every call.

Reply:
But it is not necessary when using the `crypto` library.
So we should only do it when using `pyaes`

Reply:
Sorry for the late reply, it's taken me a while to get around to testing it.

The exact same problem occurs if you don't re-initialise the AES context when using the `crypto` module. This makes sense because the issue is (most likely) in the device's handling of AES-CBC rather than any particular library.